docs/proposals/scheduling/20230707-eviction-arbitrate-mechanism-in-descheduler.md

```diff
@@ -0,0 +1,223 @@
+
+### Goals
+
+1. Defines an Arbitrate Mechanism Configuration, through which user can configure the arbitration mechanism.
+2. Describe a simple central flow control mechanism to limit the number of Pod migrations over a period of time.
+3. Describe in detail the design details of the arbitrate mechanism.
+
+### Non-Goals/Future Work
+
+1. A new CRD or Controller.
+2. The generation process of PodMigrationJob.
+3. The specific execution process of PodMigrationJob.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+The descheduler evicts Pods with different priorities, and the importance of high priority Pods is also high. Therefore, The cost of migrating high priority Pods is greater than that of low priority Pods. So users expect low-priority Pods to be executed first to minimize migration costs.
+
+#### Story 2
+
+Multiple Pods of different workloads may be concurrently evicted by the descheduler. We hope that the migrated Pods can be dispersed across different workloads (like deployments) as much as possible, in order to avoid a decrease in service availability caused by the migration process.
+
+#### Story 3
+
+Multiple Pods in different jobs are evicted. We should migrate all Pods from one job at a time and process different jobs in sequence to avoid a thundering herd effect.
+
+### Implementation Details/Notes/Constraints
+
+#### PodMigrationJob Controller
+
+The **PodMigrationJobController** will evaluate all PodMigrationJobs and select a batch of PodMigrationJob and add them to the reconcile queue for executing. This selection process is called the **arbitration mechanism**. The arbitration mechanism includes three stages: `Sort`, `GroupFilter` and `Select`.
+
+![image](/docs/images/arbitration-mechanism-design.svg)
+
+##### Controller Process
+
+Arbitration Mechanism will put arbitrated MigrationJob into the reconcile queue To achieve rate limit.
+
+The controller process can be roughly divided into three parts, namely the Controller Reconcile Process, Arbitration Process, and Migration Process:
+
+- **Controller Reconcile Process**: This is the `Reconcile` funtion of PodMigrationJob Controller, which aims to put PodMigrationJobs whose phase if `pending` or empty to the arbitration collection.
+- **Arbitration Process**: This process periodically arbitrates the PodMigrationJobs in the arbitration collection and places the PodMigrationJobs selected by the arbitration into the rate limit queue.
+- **Migration Process**: This process will read `PodMigrationJob` from the rate limit queue and call `doMigrat`e to handle the `PodMigrationJob`.
+
+###### EventHandle Process in Watch
+
+1. Intercept all Create events and place them in the arbitration collection.
```

Comment:
This section looks a little strange. I guess you want to express: when the PMJ Controller receives a PMJ event, it will directly enter the work queue instead of entering the arbitration queue, and then deliver the selected PMJ to the work queue after arbitration.  For this section, I suggest naming it `About arbitration queue`.

Reply:
Yes, that's what I mean. I'll take your advice.